starlette/testclient.py

```diff
@@ -129,6 +129,13 @@ def send(  # type: ignore
         }
 
         async def receive() -> Message:
+            nonlocal request_complete, response_complete
+
+            if request_complete:
+                while not response_complete:
+                    await asyncio.sleep(0.0001)
```

Comment:
Any hint as to why this `while` loop is required since entering this block ensures that `request_complete` is true? :)

Reply:
Ignore this comment, I didn't see `request_complete` vs `*response*_complete`. 🤦‍♂️